src/qibo/backends/numpy.py

```diff
@@ -114,15 +114,15 @@ def matrix_parametrized(self, gate):
 
     def matrix_fused(self, fgate):
         rank = len(fgate.target_qubits)
-        matrix = np.eye(2**rank, dtype=self.dtype)
+        matrix = np.eye(2**rank, dtype=np.complex128)
```

Comment:
I did not test this out, but my impression is that something like
```python
        matrix = self.np.eye(2**rank, dtype=self.dtype)
```

would work. Or maybe:
```python
        matrix = self.cast(np.eye(2**rank, dtype=complex), dtype=self.dtype)
```